File: 프로그래머스/1/64061. 크레인 인형뽑기 게임/크레인 인형뽑기 게임.js
function solution(board, moves) {
    let l=board.length;
    cboard=Array.from({ length: l }, () => Array(l).fill(0));
    for(let i=0;i<l;i++){
        for(let j=0;j<l;j++){
            cboard[i][j]=board[l-j-1][i];
        }
    }
    cboard=cboard.map(l=>l.filter(item=>item!=0));
    stack=[]
    result=0
    for(index of moves){
        row=cboard[index-1];
        if(row.length!=0){
            if(stack!=[] &&stack[stack.length-1]==row[row.length-1]){
                stack.pop();
                result+=2;
            }
            else{
                stack.push(row[row.length-1]);
            }
            cboard[index-1].pop();
        }
    }
    return result;
}

File: 프로그래머스/1/64061. 크레인 인형뽑기 게임/크레인 인형뽑기 게임.py
def solution(board, moves):
    l=len(board)
    result_board=[]
    result=0
    cboard=[[0 for _ in range(l)] for _ in range(l)]
    for i in range(l):
        for j in range(l):
            cboard[i][j]=board[l-j-1][i]
    cboard=[[item for item in l if item!=0] for l in cboard]
    for item in moves:
        a=cboard[item-1]
        if a!=[]:
            if result_board!=[] and result_board[-1]==a[-1]:
                result_board.pop()
                result+=2
            else:
                result_board.append(a[-1])
            cboard[item-1].pop()
    return result